### PR TITLE
44 search api

### PR DESCRIPTION
**Pull request by piyushtiw**

### CODE REVIEW

**Comment by piyushtiw** [comment]
@vipulnsward _a

File: Gemfile [hunk 1]
```diff
@@ -49,6 +49,12 @@ gem "bootsnap", ">= 1.1.0", require: false
 # Making it easy to serialize models for client-side use
 gem "active_model_serializers", "~> 0.9.4"
 
+#Intelligent search made easy with Rails and Elasticsearch
+gem 'searchkick'
+gem 'elasticsearch-model'
+gem 'elasticsearch-rails'
```
Comment (vipulnsward) [inline Gemfile]:
```
+gem 'elasticsearch-model'
+gem 'elasticsearch-rails'
```
Why are these dependencies needed? 
Doesn't just `searchkick` work?

Comment (piyushtiw) [reply Gemfile]: Yes no need of these gems explicitly, I removed them.

**Comment by piyushtiw** [comment]
@vipulnsward _a

File: config/initializers/elasticsearch.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+if Rails.env == "production" || Rails.env == "staging"
+  url = ENV["BONSAI_URL"]
+  transport_options = { request: { timeout: 250 } }
+  options = { hosts: url, retry_on_failure: true, transport_options: transport_options }
+  Searchkick.client = Elasticsearch::Client.new(options)
+end
```
Comment (vipulnsward) [inline config/initializers/elasticsearch.rb]: Please fix your editor for EOF

File: test/models/article_test.rb [hunk 3]
```diff
@@ -16,4 +16,24 @@ def test_article_validation
...
+
+  def test_search
+    c1 = Category.create! name: "Code"
+    org = Organization.create! name: "Google"
+    c1.articles.create!(
+      title: "How do I put nodejs code in my website?",
+      desc: "coming soon",
+      organization_id: org.id
+    )
+    Article.search_index.refresh
```
Comment (vipulnsward) [inline test/models/article_test.rb]: @piyushtiw please note that I asked this in PR elsewhere, lets not repeat feedback on every PR.

Give a newline after every logical block in test.  Example 
```
Setup code
<blank line>
Run actual code to be tested
<blank line>
Assertions
<blank line>
```

In this case, please add a blank line here

File: test/test_helper.rb [hunk 4]
```diff
@@ -30,6 +30,11 @@ class TestCase
     # in alphabetical order.
     fixtures :all
 
+    # reindex models
+    Article.reindex
```
Comment (vipulnsward) [inline test/test_helper.rb]: We don't need to reindex here, it can be done where ever its needed

**Pull request merged**